eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnifiedMap.java

```diff
@@ -263,15 +263,15 @@ public UnifiedMap(Pair<K, V>... pairs)
     }
 
     @Override
-    public MutableMap<K, V> newEmpty()
+    public UnifiedMap<K, V> newEmpty()
     {
         return new UnifiedMap<>();
     }
 
     @Override
-    public MutableMap<K, V> newEmpty(int capacity)
+    public <RK, RV> UnifiedMap<RK, RV> newEmpty(int capacity)
     {
-        return UnifiedMap.newMap(capacity);
+        return new UnifiedMap<>(capacity, this.loadFactor);
```

Comment:
@motlin Actually I changed this more out of consistency in mind. We can keep the return type the same (MutableMap) and cast it to UnifiedMap when we call it internally.

Reply:
@gjajoo Changing the return type to a subtype is probably fine. It's source compatible anyway, I'd have to check if it's also binary compatible. I meant adding generic types to an existing method isn't backwards compatible. I like the idea, just wanted to highlight the change since we're very careful about compatibility, and gather breaking changes in major version releases.